### PR TITLE
WIP [pigeon] Add sendable support for #140439

### PR DESCRIPTION
**Pull request by prasadsunny1**
[Fixes 140439](https://github.com/flutter/flutter/issues/140439)

*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

*List which issues are fixed by this PR. You must list at least one issue.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by tarrinneal** [comment]
@prasadsunny1 Are you planning on finishing the rest of the items in the check list for this pr?

**Comment by prasadsunny1** [comment]
hi @tarrinneal 
I have asked the issue author to see if this change is what they need before I move forward with writing tests.

Need help:
I am not able to figure out how and which tests to run for pigeon package. 
And what test will I need to write with my change?
Should I change the patch version or the minor version?
Where should I update the changelog?

Can you point me to a PR which demonstrates some of these things?

**Comment by stuartmorgan** [comment]
> I am not able to figure out how and which tests to run for pigeon package.

See https://github.com/flutter/packages/blob/main/packages/pigeon/CONTRIBUTING.md#testing-overview for an overview of tests; they are runnable via `tool/test.dart` (`dart run tool/test.dart [options]`).

From the CI output, at least `ios_swift_unittests` and `dart_unittests` are broken currently.

> Should I change the patch version or the minor version?

New features bump the minor version.

File: packages/pigeon/lib/swift_generator.dart [hunk 1]
```diff
@@ -125,7 +125,7 @@ class SwiftGenerator extends StructuredGenerator<SwiftOptions> {
         indent, classDefinition.documentationComments, _docCommentSpec,
         generatorComments: generatedComments);
 
-    indent.write('struct ${classDefinition.name} ');
+    indent.write('struct ${classDefinition.name}: Sendable ');
```
Comment (hellohuanlin) [inline packages/pigeon/lib/swift_generator.dart:128]: Can you create a dart annotation to gate this? this will likely crash older swift versions

**Comment by jmagman** [comment]
@prasadsunny1 Friendly ping, are you intending to follow up on the requested changes?  Thank you for your contribution!

**Comment by prasadsunny1** [comment]
> @prasadsunny1 Friendly ping, are you intending to follow up on the requested changes? Thank you for your contribution!

Hi @jmagman I would like to pass this on to other contributors as I am not able to get time to see this through properly.

**Comment by stuartmorgan** [comment]
Closing given the above; I'll leave a comment in the issue indicating that there was an in-progress PR.

**Pull request closed without merging**